io/disk/vfc-tests/vfc_dlpar.py

```diff
@@ -0,0 +1,367 @@
+        on the operation
+        '''
+        if operation == 'add':
+            cmd = 'chhwres -r virtualio -m %s -o a -p %s --rsubtype fc -s %s \
+                   -a "adapter_type=client,remote_lpar_name=%s, \
+                   remote_slot_num=%s,\\"wwpns=%s\\""' \
+                   % (self.server, self.lpar, vfc_dic["c_slot"],
+                      vfc_dic["r_lpar"], vfc_dic["r_slot"], vfc_dic["wwpn"])
+        else:
+            cmd = 'chhwres -r virtualio -m %s -o r -p %s  -s %s' \
+                   % (self.server, self.lpar, vfc_dic["c_slot"])
+        try:
+            self.run_command(cmd)
+        except CommandFailed as cmdfail:
+            self.log.debug(str(cmdfail))
+            self.fail("Network virtualization %s device operation \
+                       failed" % operation)
+        time.sleep(self.opp_sleep_time)
+        self.drc_name_verification_hmc(operation, vfc_dic["c_slot"])
+        self.linux_name_verification_host(operation, vfc_dic["drc"])
+        self.mpath_verification(operation, vfc_dic["paths"],
+                                vfc_dic["drc"])
+
+    def drc_name_verification_hmc(self, operation, c_slot):
+        '''
+        verify the vfc slot/rdc_name exists in HMC
+        '''
+        err_slot = []
+        drc_name = self.get_drc_name(c_slot)
+        if operation == "add":
+            if not drc_name:
+                err_slot.append(c_slot)
+        elif operation == "remove":
+            if drc_name:
+                err_slot.append(c_slot)
+        else:
+            self.fail("mis-match operation name: it should be add or remove")
+
+        if err_slot:
+            self.fail("HMC verifction fail for %s: %s" % (drc_name, operation))
+        else:
+            self.log.info("HMC verfction succes %s:%s" % (drc_name, operation))
+
+    def linux_name_verification_host(self, operation, drc_name):
+        '''
+        verify the linux_name or drc_name in host
+        '''
+        err_slot = []
+        linux_name = self.get_linux_name_drc(drc_name)
+        if operation == "add":
+            if not linux_name:
+                err_slot.append(drc_name)
+        elif operation == "remove":
+            if linux_name:
+                err_slot.append(drc_name)
+        else:
+            self.fail("mis-match operation name: it should be add or remove")
+
+        if err_slot:
+            self.fail("Host verifction fail for %s:%s" % (drc_name, operation))
+        else:
+            self.log.info("Host verfction suces %s:%s" % (drc_name, operation))
+
+    def mpath_verification(self, operation, paths, drc):
+        '''
+        verify the paths status on add or remove operations of vfc
+        '''
+        err_paths = []
+        if operation == "add":
+            state1 = "active"
+            state2 = "ready"
+        elif operation == "remove":
+            state1 = "failed"
+            state2 = "faulty"
+        else:
+            self.fail("invalid operation : %s", operation)
```

Comment:
return instead of fail

Reply:
@Naresh-ibm I see in many places you are validating the input of functions. It is a good practice for functions which are used in multiple places, it is over complication in this context. Reason: Only this script calls it, and the input you are validating is `operation`, which you are controlling directly from calling statement.

Also, whenever you want to do this validation, you should return to the calling statement, which should only take actions such as pass / fail / error.

Reply:
I have used single function for both remove and add operation. thats the reason, I have used this validation. the last else part for my reference I have used. we can remove it. but rest need to be there.